analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -375,52 +376,63 @@ public void Finally_WrappedInLocalLifetimeRegion()
         public void Finally_ThrowInTry()
         {
             const string code = @"
-Tag(""BeforeTry"");
+var tag = ""BeforeTry"";
 try
 {
     Tag(""InTry"");
     throw new System.Exception();
-    Tag(""UnreachableInTry"");
+    tag = ""UnreachableInTry"";
 }
 finally
 {
-    Tag(""InFinally"");
+    tag = ""InFinally"";
 }
-Tag(""UnreachableAfterFinally"");";
-            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+tag = ""UnreachableAfterFinally"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
                 "BeforeTry",
                 "InTry",
-                "InFinally");
+                "InFinally",                // With Exception thrown by Tag("InTry")
+                "InFinally");               // With Exception thrown by `throw`
+
+            validator.TagStates("InFinally").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasUnknownException(x))
+                     .And.ContainSingle(x => HasExceptionOfTypeException(x));
         }
 
         [TestMethod]
         public void Finally_ThrowInFinally()
         {
             const string code = @"
-Tag(""BeforeTry"");
+var tag = ""BeforeTry"";
 try
 {
     Tag(""InTry"");
 }
 finally
 {
-    Tag(""InFinally"");
+    tag = ""InFinally"";
     throw new System.Exception();
-    Tag(""UnreachableInFinally"");
+    tag  = ""UnreachableInFinally"";
 }
-Tag(""UnreachableAfterFinally"");";
-            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+tag = ""UnreachableAfterFinally"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
                 "BeforeTry",
                 "InTry",
-                "InFinally",    // With Exception thrown by Tag("InTry")
+                "InFinally",
                 "InFinally");
+
+            validator.TagStates("InFinally").Should().HaveCount(2)
+                .And.ContainSingle(x => HasNoException(x))
+                .And.ContainSingle(x => HasUnknownException(x));
```

Comment:
This assertion repeats many times. What about method in `validator` to assert that? `validator.ValidateTagHasNoAndUnknownException("InFinally")`

Reply:
Personally, I don't like this. It's very specific. I prefer to have the assertions inline.

Reply:
These will look great once we will implement the check extensions for fluent assertions:

```
validator.TagStates("InFinally").Should().HaveCount(2)
	.And.ShouldContainSingleNoException()
	.And.ShouldContainSingleUnknownException()
```

For me, this will be a lot easier to understand.